src/shared/Util.cpp

```diff
@@ -316,6 +316,49 @@ std::string ConvertTimeStampToDataTime(uint32 timestamp)
     return szResult;
 }
 
+unsigned int TimeToGametime(time_t unixtime)
+{
+    uint32 gameTime;
+    time_t basetime = unixtime;
+
+    time_t minutes = ((unixtime / 60) % 60);
+    time_t hours = ((unixtime / (60 * 60)) % 60);
+
+    struct tm * timeinfo = localtime(&basetime);
+    uint32 DayOfTheWeek;
+    if (timeinfo->tm_wday == 0)
+        DayOfTheWeek = 6;                           //  It's Sunday
+    else
+        DayOfTheWeek = timeinfo->tm_wday - 1;       //  0b111 = (any) day, 0 = Monday ect)
+    uint32 DayOfTheMonth = timeinfo->tm_mday - 1;   //  Day - 1 (0 is actual 1) its now the 20e here.
+    uint32 CurrentMonth = timeinfo->tm_mon;         //  Month - 1 (0 is actual 1) same as above.
+    uint32 CurrentYear = timeinfo->tm_year - 100;   //  2000 + this number results in a correct value for this crap.
+
+#define MINUTE_BITMASK      0x0000003F
+#define HOUR_BITMASK        0x000007C0
+#define WEEKDAY_BITMASK     0x00003800
+#define DAY_BITMASK         0x000FC000
+#define MONTH_BITMASK       0x00F00000
+#define YEAR_BITMASK        0x1F000000
+#define UNK_BITMASK         0xE0000000
+
+#define MINUTE_SHIFTMASK    0
```

Comment:
These defined values are already in WUtil.cpp. Move it to a proper header files.